doc/host_config_schema/cchost_config.json

```diff
@@ -49,6 +49,11 @@
                 "default": "Value of 'bind_address'",
                 "description": "The published RPC address advertised to clients"
               },
+              "protocol": {
+                "type": "string",
+                "default": "(empty)",
+                "description": "The RPC protocol used (udp | tcp)"
+              },
```

Comment:
I think this should be an enum (see for example `"endorsement"` on line `62`) that defaults to TCP.

Reply:
I also wonder whether the protocol field should be more opinionated with regards to the application layer protocol too, i.e. `http/1`, `http/2` or `http/3`?

Reply:
Right now, this is at best experimental. It's not clear to me what's the best granularity (http, quic, udp, tcp) or if we want a parseable descriptor (like the "local,tcp" hack in the string representation). I personally prefer to leave as a string until we're sure what we want and how we want.

I'm also not sure about being too opinionated. HTTP 1 through 2 can really only work with TCP (+TLS) and HTTP 3 can only work with QUIC and QUIC can only work with UDP (no TLS). And for all purposes, HTTP 2 is an extension of 1.1 which is an extension of 1, so we end up with only two choices:
 * HTTP {1, 1.1, 2} + TCP + TLS or
 * HTTP 3 + QUIC

Whether we call it QUIC or UDP or HTTP 3 doesn't make much of a difference, I think.

Reply:
Well summarised. We may also want to distinguish http from httpS (in case we "add" support for this in the future?). I think there's an argument for only allowing HTTP/2 too (e.g. when apps are gRPC only and we don't want to support the upgrade from 1.1 to 2).

Reply:
I thought we only wanted to support TLS protected streams, be it HTTPS or QUIC (ie, no HTTP).

About gRPC, seems it makes use of core ideas in HTTP/2 that aren't in HTTP/1.1, so we'll have to differentiate that, too. In a way, this maps nicely to `{ http1, http2, http3 }` better than `{ tcp, udp }`.

But for now, since even calling it QUIC is a misnomer, we can leave it as UDP because that's all it is. We can easily change the naming later or create an enum or an integer `{ 1, 2, 3 }`, once we have an HTTP/3 endpoint.